tss/node/signer/verify.go

```diff
@@ -129,3 +155,24 @@ func (p *Processor) CacheVerify(key string, value bool) bool {
 	defer p.cacheVerifyLock.Unlock()
 	return p.cacheVerify.Set(key, value)
 }
+
+func (p *Processor) GetVerify(key string) (bool, bool) {
+	p.cacheVerifyLock.RLock()
+	defer p.cacheVerifyLock.RUnlock()
+	return p.cacheVerify.Get(key)
+}
+
+func group(slice [][32]byte, segment int) map[int][][32]byte {
+	size := len(slice)
+	var segments = make(map[int][][32]byte)
+	quantity := size / segment
+	remainder := size % segment
+	i := 0
+	for ; i < quantity; i++ {
+		segments[i*segment] = slice[i*segment : (i+1)*segment]
```

Comment:
would it be better to add a  slice length check?

Reply:
There is no array out of bounds